src/test/java/gameframework/game/GameUniverseDefaultImplTest.java

```diff
@@ -0,0 +1,118 @@
+package gameframework.game;
+
+import static org.junit.Assert.*;
+
+import java.util.Iterator;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import gameframework.game.mocks.GameDataMock;
+import gameframework.game.mocks.GameEntityMock;
+import gameframework.game.mocks.GameMovableEntityMock;
+import gameframework.game.mocks.GameMoveBlockerEntityMock;
+import gameframework.game.mocks.GameOverlappableEntityMock;
+import gameframework.motion.blocking.MoveBlockerChecker;
+import gameframework.motion.overlapping.mocks.OverlapProcessorDefaultImplMock;
+
+public class GameUniverseDefaultImplTest {
+
+	protected GameUniverseDefaultImpl gameUniverseDefaultImpl;
+	protected MoveBlockerChecker moveBlockerChecker;
+	
+	@Before
+	public void init() {
+		gameUniverseDefaultImpl = new GameUniverseDefaultImpl();
+		gameUniverseDefaultImpl.setGameData(new GameDataMock());
+
+		moveBlockerChecker = gameUniverseDefaultImpl.getMoveBlockerChecker();
+	}
+
+	public Boolean entityExist(Iterator<?> i, GameEntity entity) {
+		while(i.hasNext()) {
+			if(entity.equals(i.next())) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	public Boolean hasGameEntity(GameEntity entity) {
+		return entityExist(gameUniverseDefaultImpl.getGameEntitiesIterator(), entity);
+	}
+
+	public Boolean moveBlockerEntityAdded(GameEntity entity) {
```

Comment:
misnamed method. It does not test if an entity has been added. This method tests if an entity is present.